packages/trpc/server/routers/viewer/webhook/util.ts

```diff
@@ -10,41 +11,132 @@ export const webhookProcedure = authedProcedure
   .use(async ({ ctx, input, next }) => {
     // Endpoints that just read the logged in user's data - like 'list' don't necessary have any input
     if (!input) return next();
-    const { eventTypeId, id } = input;
+    const { id, teamId, eventTypeId } = input;
 
-    // A webhook is either linked to Event Type or to a user.
-    if (eventTypeId) {
-      const team = await prisma.team.findFirst({
+    if (id) {
+      //check if user is authorized to edit webhook
+      const webhook = await prisma.webhook.findFirst({
         where: {
-          eventTypes: {
-            some: {
-              id: eventTypeId,
-            },
-          },
+          id: id,
         },
         include: {
-          members: true,
+          user: true,
+          team: true,
+          eventType: true,
         },
       });
 
-      // Team should be available and the user should be a member of the team
-      if (!team?.members.some((membership) => membership.userId === ctx.user.id)) {
-        throw new TRPCError({
-          code: "UNAUTHORIZED",
-        });
+      if (webhook) {
+        if (webhook.teamId) {
+          const user = await prisma.user.findFirst({
+            where: {
+              id: ctx.user.id,
+            },
+            include: {
+              teams: true,
+            },
+          });
+
+          if (
+            user &&
+            !user.teams.some(
+              (membership) =>
+                membership.teamId === webhook.teamId &&
+                (membership.role === MembershipRole.ADMIN || membership.role === MembershipRole.OWNER)
+            )
+          ) {
```

Comment:
We can split this validations outside the if() and then just call if(userBelongsToTeam && userIsAdminOrOwner) { ... }